src/types/Health.ts

```diff
@@ -87,22 +101,67 @@ interface ThresholdStatus {
 const RATIO_NA = -1;
 
 export const ratioCheck = (availableReplicas: number, currentReplicas: number, desiredReplicas: number): Status => {
-  // No Pods returns No Health Info
-  if (desiredReplicas === 0 && currentReplicas === 0) {
-    return NA;
+  /*
+    IDLE STATE
+ */
+  // User performed the action
+  if (desiredReplicas === 0) {
+    return IDLE;
   }
-  // No available Pods when there are desired and current means a Failure
+
+  /*
+   DEGRADED STATE
+  */
+  if (
+    desiredReplicas > 0 &&
+    currentReplicas > 0 &&
+    availableReplicas > 0 &&
+    (currentReplicas < desiredReplicas || availableReplicas < desiredReplicas)
+  ) {
+    return DEGRADED;
+  }
+
+  /*
+     FAILURE STATE
+  */
+  if (desiredReplicas > 0 && (currentReplicas === 0 || availableReplicas === 0)) {
+    return FAILURE;
+  }
+
+  // Some pods are up but not ready
   if (desiredReplicas > 0 && currentReplicas > 0 && availableReplicas === 0) {
     return FAILURE;
   }
+
+  if (desiredReplicas > 0 && currentReplicas < desiredReplicas) {
+    return FAILURE;
+  }
+
+  if (desiredReplicas === currentReplicas && availableReplicas > currentReplicas) {
```

Comment:
I think this is not going to happen: `currentReplicas > availableReplicas` always. As "availableReplicas" are those pods that are up and are responding, meanwhile "currentReplicas are all pods up".

It sounds a corner case that won't never happen (unless I'm missing something).